hw/opentitan/ot_otp_dj.c

```diff
@@ -1522,100 +1437,117 @@ static uint64_t ot_otp_dj_get_buffered_part_digest(OtOTPDjState *s, int part)
 
 static bool ot_otp_dj_is_part_digest_offset(int part, hwaddr addr)
 {
-    uint16_t offset = ot_otp_dj_get_part_digest_offset(part);
+    uint16_t offset = OtOTPPartDescs[part].digest_offset;
 
     return (offset != UINT16_MAX) && ((addr & ~OTP_DIGEST_ADDR_MASK) == offset);
 }
 
+static bool ot_otp_dj_is_part_zer_offset(int part, hwaddr addr)
+{
+    uint16_t offset = OtOTPPartDescs[part].zer_offset;
+
+    return (offset != UINT16_MAX) && ((addr & ~OTP_ZER_ADDR_MASK) == offset);
+}
+
+static uint32_t ot_otp_dj_get_part_digest_reg(OtOTPDjState *s, uint32_t offset)
+{
+    /*
+     * Offset it the register offset from the first read digest 32-bit register.
+     * all digest are 64-bits, which means each partitition have two 32-bit
+     * registers to expose their digest.
+     *
+     * Not all partitions have digests, which means the offset argument is the
+     * relative partition offset for those partitions that features a digest, as
+     * there is no defined digest access registers defined for partitions that
+     * do not have a digest.
+     *
+     * Need to traverse the partition table to only account for those partitions
+     * with a digest to match the proper offset.
+     */
+    unsigned part_look_ix = (int)offset >> 1u;
+    bool hi = (bool)(offset & 0x1);
+
+    unsigned part_ix = 0;
+    for (unsigned part_dig_ix = 0; part_ix < OTP_PART_COUNT; part_ix++) {
```

Comment:
Should these be swapped? (functionally the same, but less confusing...)
```suggestion
    unsigned part_dig_ix = 0;
    for (unsigned part_ix = 0; part_ix < OTP_PART_COUNT; part_ix++) {
```

Reply:
`part_ix` is required outside the `for` block, while `part_dig_ix` is local to the loop, so I'm not sure how to rewrite it.